### PR TITLE
Add "Select All" action to multiSelect

### LINKED ISSUE #895: Add "select all" to selection menu
After pressing and holding a media entry the selection menu should include "select all" so one does not have to individually select several additional entries.

image

### PR DESCRIPTION
**Pull request by marius-luca-87**
**Changes**

Adds a "Select all" helper to the multi-select menu.

image

**Issues**
Fixes #895

### CODE REVIEW

**Review by ferferga — APPROVED** [review]

File: src/components/multiSelect/multiSelect.js [hunk 1]
```diff
@@ -246,6 +252,21 @@ import itemHelper from '../itemHelper';
...
 
                             switch (id) {
+                                case 'selectall':
+                                    {
+                                        const elems = document.querySelectorAll('.itemSelectionPanel');
+                                        for (let i = 0, length = elems.length; i < length; i++) {
+                                            const chkItemSelect = elems[i].querySelector('.chkItemSelect');
+
+                                            if (chkItemSelect) {
+                                                if (!chkItemSelect.classList.contains('checkedInitial') && !chkItemSelect.checked) {
```
Comment (thornbill) [inline src/components/multiSelect/multiSelect.js:262]: I would combine these two if statements.

**Review by thornbill — CHANGES_REQUESTED** [review]

**Comment by thornbill** [comment]
I noticed an unusual behavior when testing this. In this screen there should be 100 items shown and selected, but for some reason there appears to be 226 selected.

all selelcted

**Comment by marius-luca-87** [comment]
Apparently a few cards are left hidden and the way the current system works is by assuming you will never be able to click them.
image
The only quick way I could find would be to check for "chkItemSelect.getBoundingClientRect().width != 0" before selecting them.

**Review by thornbill — APPROVED** [review]
Well that is not great... it definitely shouldn't be enabling the multiselect stuff on hidden pages or inactive tabs, but that doesn't relate to your changes really. I think your work around for that is fine. I will see if I can find a fix to prevent multiselect from being enabled on the hidden views after this is merged. :sweat_smile:

**Pull request merged**